src/main/java/com/dev/cinema/security/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,36 @@
+package com.dev.cinema.security;
+
+import com.dev.cinema.exception.AuthenticationException;
+import com.dev.cinema.lib.Inject;
+import com.dev.cinema.lib.Service;
+import com.dev.cinema.model.User;
+import com.dev.cinema.service.UserService;
+import com.dev.cinema.util.HashUtil;
+import java.util.Optional;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> userFindByEmail = userService.findByEmail(email);
+        if (userFindByEmail.isPresent()) {
+            User user = userFindByEmail.get();
+            String hashPassword = HashUtil.hashPassword(password, user.getSalt());
+            if (hashPassword.equals(user.getPassword())) {
+                return user;
+            }
+        }
+        throw new AuthenticationException("Can't get login by email " + email);
```

Comment:
let's make the message more informative
```suggestion
        throw new AuthenticationException("The login or password is incorrect");
```

Reply:
fixed